src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3608,6 +3609,21 @@ void Creature::InitializeInteractSpellId()
         SetInteractSpellId(0);
 }
 
+void Creature::InitializeSummonInfo()
+{
+    _summonInfo = std::make_unique<SummonInfo>(this);
```

Comment:
shouldn't be required, because it's assumed that the void will be called just once, but shouldn't we check for it being null?

`
if (!_summonInfo)
        _summonInfo = std::make_unique<SummonInfo>(this);
`

Reply:
I would rather throw an assert instead to directly tell the user that this method is not meant to be called twice.

Reply:
yeah that would be nice as well, if filled, an assert instead